usvm-ts/src/main/kotlin/org/usvm/util/TsImports.kt

```diff
@@ -0,0 +1,203 @@
+package org.usvm.util
+
+import org.jacodb.ets.model.EtsExportInfo
+import org.jacodb.ets.model.EtsExportType
+import org.jacodb.ets.model.EtsFile
+import org.jacodb.ets.model.EtsImportInfo
+import org.jacodb.ets.model.EtsImportType
+import org.jacodb.ets.model.EtsScene
+
+sealed class ImportResolutionResult {
+    data class Success(val file: EtsFile) : ImportResolutionResult()
+    data class NotFound(val reason: String) : ImportResolutionResult()
+}
+
+sealed class SymbolResolutionResult {
+    data class Success(val file: EtsFile, val exportInfo: EtsExportInfo) : SymbolResolutionResult()
+    data class FileNotFound(val reason: String) : SymbolResolutionResult()
+    data class SymbolNotFound(val file: EtsFile, val symbolName: String, val reason: String) : SymbolResolutionResult()
+}
+
+fun EtsScene.resolveImport(
+    currentFile: EtsFile,
+    importPath: String,
+): ImportResolutionResult {
+    return try {
+        when {
+            importPath.startsWith("@") -> resolveSystemLibrary(importPath)
+            importPath.startsWith("/") -> resolveAbsolutePath(importPath)
+            else -> resolveRelativePath(currentFile, importPath)
+        }
+    } catch (e: Exception) {
+        ImportResolutionResult.NotFound(e.message ?: "Unknown error")
+    }
+}
+
+fun EtsScene.resolveSymbol(
+    currentFile: EtsFile,
+    importPath: String,
+    symbolName: String,
+    importType: EtsImportType,
+): SymbolResolutionResult {
+    return when (val fileResult = resolveImport(currentFile, importPath)) {
+        is ImportResolutionResult.NotFound -> SymbolResolutionResult.FileNotFound(fileResult.reason)
+        is ImportResolutionResult.Success -> resolveSymbolInFile(fileResult.file, symbolName, importType)
+    }
+}
+
+fun EtsScene.resolveImportInfo(
+    currentFile: EtsFile,
+    importInfo: EtsImportInfo,
+): SymbolResolutionResult {
+    return resolveSymbol(currentFile, importInfo.from, importInfo.originalName, importInfo.type)
+}
+
+private fun EtsScene.resolveSystemLibrary(importPath: String): ImportResolutionResult {
+    val foundFile = sdkFiles.find { file ->
+        file.signature.fileName.equals(importPath, ignoreCase = true) ||
```

Comment:
with (field.signature.filename)

Reply:
Do not over-complicate things with those context functions...